src/js/client/ajaxCapable.js

```diff
@@ -0,0 +1,138 @@
+/*
+
+ A component to standardize the process of retrieving data and then updating the model when the results are received.
+
+ The component sends a request when its `makeRequest` invoker is called.  The request parameters are:
+
+   1.  The request data, which is the result of transforming the model according to the rules specified in
+       `options.rules.modelToRequestPayload`.
+
+   2.  The ajax options, which are the result of transforming `options.ajaxOptions` according to the rules specified in
+       `options.rules.ajaxOptions`.
+
+       The end result is expected to contain options in the format used by `jQuery.ajax()`,
+       with the exception of the `json: true` option, which is a nicety to improve handling of JSON data payloads.
+
+       The rules used to prepare the final AJAX options are expanded before each request, so that you can use IoC
+       references that refer to the current state of the component.
+
+ Once the request is sent, the following workflow applies:
+
+   1.  If the AJAX request returns an error, the `handleError` invoker is called.  The error is transformed
+       using the rules found in `options.rules.errorResponseToModel`, and the results are applied to the component's
+       model using the change applier.
+
+   2.  If the AJAX request is successful, the `handleSuccess` invoker is called.  The response data is transformed
+       using the rules found in `options.rules.successResponseToModel`, and applied to the model using the change
+       applier.
+
+ All model changes are batched, i.e. the entire set of changes results in a single transaction.
+
+ This component does not handle any rendering, you are expected to do that yourself, or use a grade that handles that.
+
+ */
+// TODO:  Reconcile this with the larger migration to dataSources.
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.ajaxCapable");
+
+    // Apply all changes in a single transaction.  Also ensures that values flagged with `null` are deleted from the model.
+    gpii.templates.ajaxCapable.batchChanges = function (that, changeSet) {
```

Comment:
The main use case is to make it possible to indicate that model material should be removed based on material received from a REST call.  Another use case that's purely handled in transforms is removing a success message on error, or vice versa.

We have to specifically check for `null` because keys that resolve to `undefined` values are stripped by `fluid.model.transformWithRules`, while keys that resolve to `null` values are not.

This seemed to me like a more natural way to support removing material based on an update than requiring everyone to follow some convention like setting the value to `DELETE`.

If you have another idea about how to handle this, please let me know.